game-core/src/main/java/games/strategy/net/MacFinder.java

```diff
@@ -59,7 +60,7 @@ public static String getHashedMacAddress(final byte[] macAddressBytes) {
   }
 
   private static String getHashedMacAddress(final String macAddress) {
-    return games.strategy.util.Md5Crypt.crypt(macAddress, HASHED_MAC_ADDRESS_SALT);
```

Comment:
It was intentional to use the fully-qualified class name for `Md5Crypt` to reduce the number of deprecation warnings we get during the Gradle build.

Reply:
Okay, we were missing a comment then, no way to know about that just looking at the code.

Reply:
Wait.. we still get a deprecation warning using a fully qualified package name or not. I certainly still see one.

I think I support the original change here:
- fully qualified package name should have no impact on deprecation usage
- even if it did, hard to not call that hackery, the warning is still legit as we're using a deprecated class.

Given that, it does not seem the fully qualified package name should have any bearing on the deprecation warning. (and even if it did, it would still be the right thing to do rather than sweeping a valid warning under the rug)

Reply:
Spotted a second usage in the same file, not fully qualified (line ~309)
```
    try {
      return HASHED_MAC_ADDRESS_SALT.equals(Md5Crypt.getSalt(value));
    } catch (final IllegalArgumentException e) {
      return false;
    }
```
I very much disagree adding fully qualified package names to try and remove deprecation warnings. It's not a fix, and it's the wrong fix even if it were. We were not even consistent about it either.

Reply:
Yes, what @RoiEXLab said: the intention was to _reduce_ not _eliminate_.  Gradle build output is extremely difficult to parse in real-time with so many deprecation warnings hiding other, potentially more important, messages.

It was suggested about a month ago that we simply un-deprecate `Md5Crypt`.  My response at the time was to leave it alone because we were anticipating an incompatible release in which we could remove `Md5Crypt` usage.  An incompatible release no longer appears to be around the bend.

After recently doing some work in this area, I realized we'll still have a legitimate use for `Md5Crypt` related to hashed MACs, which has nothing to do with password security.  So, even after we remove the MD5-crypted passwords from the lobby, we'll still probably continue to use this class for other non-security related functionality.

So, should we simply un-deprecate `Md5Crypt`?  We can add a big fat notice in the Javadocs that states something like "do not use this class to protect passwords or other sensitive information."

Reply:
Here's the original PR for reference: #2700.

Reply:
> So, should we simply un-deprecate Md5Crypt? 

Sounds like.

>  because the import + the actual usage count as 2 deprecation usages

oh... This is is a surprise. I don't think that is an actual solution, we're making the code dirtier because of analysis linting. The benefit/cost calculation seems wrong, one extra warning out of thousands seems like a very small victory. Meanwhile we have placed a landmine in the code.

IF output is concern, IMO we could solve by tuned down the javadoc and naming of checkstyle, that is over a thousand lines of output.